app/components/chat_message/chat_message.js

```diff
@@ -35,4 +35,24 @@ export default class extends Controller {
     this.element.classList.remove('ChatMessage--expanded');
     this.toggleTarget.setAttribute('aria-expanded', 'false');
   }
+
+  copy() {
+    const text = this.textTarget.innerText;
+    const sender = this.data.get('sender-name');
+    let clipboardText = `${sender}: ${text}`;
+
+    if(!sender) {
+      clipboardText = text;
+    }
+
+    navigator.clipboard.writeText(clipboardText).then(() => this.onCopy());
+  }
+
+  onCopy() {
+    this.element.classList.add('ChatMessage--copied');
+
+    setTimeout(() => {
+      this.element.classList.remove('ChatMessage--copied');
+    }, 2000);
```

Comment:
Move this into a constant…

```suggestion
    setTimeout(() => {
      this.element.classList.remove('ChatMessage--copied');
    }, SUCCESS_MESSAGE_DURATION);
```